tests/test_allauth_2fa.py

```diff
@@ -253,6 +255,28 @@ def test_2fa_removal(client, john_with_totp, token_state):
     assert was_removed == (token_state == "correct" or token_state == "static")
 
 
+@patch.object(app_settings, "REQUIRE_OTP_ON_DEVICE_REMOVAL", new=False)
+def test_2fa_removal_without_otp(client, john_with_totp):
+    """Removing 2FA should be possible with a correct token."""
+    user, totp_device, static_device = john_with_totp
+    login(client, expected_redirect_url=TWO_FACTOR_AUTH_URL)
+    do_totp_authentication(
+        client,
+        totp_device=totp_device,
+        expected_redirect_url=settings.LOGIN_REDIRECT_URL,
+    )
+    assert user.totpdevice_set.exists()
+
+    # Navigate to 2FA removal view
+    client.get(reverse("two-factor-remove"))
+    # ... and POST to confirm
+    client.post(reverse("two-factor-remove"), {})
+
+    # The only case when the TOTP device should be removed is when the token is correct.
```

Comment:
This copy-pasted comment is wrong for this test. You could combine this with the above test using `pytest.mark.parametrize` anyhow –

```python
@pytest.mark.parametrize("require_token", [False, True])
def test_2fa_removal(client, john_with_totp, token_state, require_token, monkeypatch):
    monkeypatch.setattr(app_settings, "REQUIRE_OTP_ON_DEVICE_REMOVAL", require_token)
```
and then at the very end verify that if `require_token` is false, the token is removed no matter the user input?

Reply:
Thank you, that makes sense!
I will adjust it later